File: pins.ts
namespace pins {

   /**
    * Configure the touch detection for the pins and logo.
    * P0, P1, P2 use resistive touch by default.
    * The logo uses capacitative touch by default.
    * @param name target to change the touch mode for, eg: TouchTarget.P0
    * @param mode the touch mode to use, eg: TouchTargetMode.Capacitative
    */
    //% help=pins/touch-set-mode weight=60
    //% blockId=device_touch_set_type block="set %name to touch mode %mode"
    //% shim=pins::touchSetMode
    export function touchSetMode(name: TouchTarget, mode: TouchTargetMode): void {
        return // required for noop
    }

}

File: pins.cpp
#include "pxt.h"

enum class TouchTargetMode {
    //% block="resistive"
    Resistive = 0,
    //% block="capacitative"
    Capacitative = 1
};

// Maybe we can add to TouchPin when in pxt-microbit? Would need error handing elsewhere.
// TouchTargetMode -> TouchPinMode if we do that.
enum class TouchTarget {
    //% block="P0"
    P0 = MICROBIT_ID_IO_P0,
    //% block="P1"
    P1 = MICROBIT_ID_IO_P1,
    //% block="P2"
    P2 = MICROBIT_ID_IO_P2,
    //% block="logo"
    LOGO = 121 // Should be MICROBIT_ID_FACE after dal.d.ts update.
};

namespace pins {

//%
void touchSetMode(TouchTarget name, TouchTargetMode mode) {
    const auto pin = name == TouchTarget::LOGO ? &uBit.io.face : getPin((int)name);
    if (pin) {
        pin->isTouched(mode == TouchTargetMode::Capacitative ? TouchMode::Capacitative : TouchMode::Resistive);
    }
}

}
